stl/inc/memory

```diff
@@ -2351,6 +2369,25 @@ void _Uninitialized_value_construct_multidimensional_n(_Ty* const _Out, const si
     }
 }
 
+template <class _Ty>
+void _Uninitialized_default_construct_multidimensional_n(_Ty* const _Out, const size_t _Size) {
+    if constexpr (!is_trivially_default_constructible_v<_Ty>) {
+        if constexpr (is_array_v<_Ty>) {
+            _Reverse_destroy_multidimensional_n_guard<_Ty> _Guard{_Out, 0};
+            for (size_t& _Idx = _Guard._Index; _Idx < _Size; ++_Idx) {
+                _Uninitialized_default_construct_multidimensional_n(_Out[_Idx], extent_v<_Ty>);
+            }
+            _Guard._Target = nullptr;
+        } else {
+            _Uninitialized_rev_destroying_backout _Backout{_Out};
+            for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+                _Backout._Emplace_back_for_overwrite();
+            }
+            _Backout._Release();

```

Comment:
Should we simplify this to
```suggestion
            _Uninitialized_default_construct_n(_Out, _Size); 
```

Reply:
There is no such function as `_Uninitialized_default_construct_n` and we can't use `uninitialized_default_construct_n` because it doesn't destroy elements in the reverse order.

Reply:
Hm there is the ranges version (where I found a unneeded specialization that I forgot to remove so thanks ;)) 

Should we also destroy in backwards order there?

Reply:
`ranges::uninitialized_default_construct_n` needs to support forward iterators and destroying in reverse order with them would be hard.